Use Global DFL

wynn1212 opened the pull request:
This pull request primarily addresses the following:
- Resolves issue #16 on the latest Steam Deck beta client and Decky Loader by implementing Global DFL.
    - Tested with Steam version: `v1713825714`.
    - Tested with Decky Loader version: `v2.12.0`.
- Bumps `decky-frontend-lib` to version `3.25.0` to resolve issue https://github.com/popsUlfr/SDH-PauseGames/pull/15#issue-2163572686

wynn1212 committed:
Update rollup.config.js to use global DFL

wynn1212 committed:
Bump decky-frontend-lib to 3.25.0

alex-f-k commented:
Can confirm, lib bump fixes at least both #16 and #21

decky lib changed `staticClasses` reference and this line https://github.com/popsUlfr/SDH-PauseGames/blob/78ab7f9c8f259ad322209823948a74e5afb0e328/src/index.tsx#L210 compiled with the old lib produced npe:
```diff
-title: window.SP_REACT.createElement("div", { className: deckyFrontendLib.staticClasses.Title }, "Pause Games"),
+title: window.SP_REACT.createElement("div", { className: staticClasses.Title }, "Pause Games"),
```

@popsUlfr when you have time, could you merge it and publish the new version to the deckbrew? Appreciate your plugin, btw, thanks!

wynn1212 commented:
Forget to mention, this patch does work on SteamOS `3.5.19`, therefore issue #21 is resolvable

popsUlfr commented:
@wynn1212 @alex-f-k Thank you so much for your efforts and sorry for my lack of reactivity.

The issue is that I'm way too out of the loop when it comes to decky and the plugins (I haven't been using decky on my deck), and I just can't invest effort into myself for the foreseeable future unfortunately.

The best course of action would be to submit your own fork of the plugin to the plugin database.
You have my approval to take over and submit your own plugin to replace this one (not sure if this means much since right now you know way more about decky than I do to be able to rectify the situation, I mean that's why I would like to delegate).

Mothamoz commented:
Hoping to see a fork soon :)

wynn1212 committed:
Bump SDH-PauseGames version to 0.4.3

wynn1212 commented:
Hi @popsUlfr, Thank you for your affirmation.

But unfortunately, I won't be able to take over this plugin as I'm not familiar with all the codes within it. Additionally, I have other important projects to do. 

When I initially encountered this issue, I just happened to found a solution in other plugins, copied it over to this one, and submitted the pull requests. Therefore, I don't believe I'll have the time to contribute further improvements or add additional features to this project. 

However, I can suggest that you add some collaborators you trust to this project, perhaps some of the past contributors. If you need the tester to test the new release, I'm also available to assist, as my SteamOS is on the preview channel.

In the meantime, I'll remain available for testing and offer any assistance and fixes to the best of my ability. Thank you for considering my input, and I wish you the best in finding a suitable path forward for the project.

wynn1212 committed:
update pnpm lockfileVersion to 6.0

aqlstar commented:
Guys, I miss this plugin so much. please update soon. Thank you all

Mothamoz commented:
@popsUlfr can you perhaps authorize someone like @wynn1212 to merge PRs for this project so it can continue to be used by many?

popsUlfr commented:
> But unfortunately, I won't be able to take over this plugin as I'm not familiar with all the codes within it. Additionally, I have other important projects to do.

No need to take over, just this one time you might try to submit the update to the plugin database by pointing to your fork if you can spare the effort. Or even someone else could do just the submitting part and point to your temporary fork.

> @popsUlfr can you perhaps authorize someone like @wynn1212 to merge PRs for this project so it can continue to be used by many?

Just merging the PR is not enough, one would need to submit a PR to the plugin database to get it "upstream". I would really prefer to be able to vet any changes to my repo myself first. Otherwise the quickest/easiest way right now is to just point to the temp fork in an update PR to the plugin database.

jackson-xp commented:
"Guys, these days without this plugin were really tough, the long loading times in games were torture! Hahahaha!"
"Could any energetic mate please help submit it to the Decky Loader database?" ：）

wynn1212 commented:
> > But unfortunately, I won't be able to take over this plugin as I'm not familiar with all the codes within it. Additionally, I have other important projects to do.
> 
> No need to take over, just this one time you might try to submit the update to the plugin database by pointing to your fork if you can spare the effort. Or even someone else could do just the submitting part and point to your temporary fork.
> 
> > @popsUlfr can you perhaps authorize someone like @wynn1212 to merge PRs for this project so it can continue to be used by many?
> 
> Just merging the PR is not enough, one would need to submit a PR to the plugin database to get it "upstream". I would really prefer to be able to vet any changes to my repo myself first. Otherwise the quickest/easiest way right now is to just point to the temp fork in an update PR to the plugin database.

Hi @popsUlfr and everyone,

I've submitted a pull request to https://github.com/SteamDeckHomebrew/decky-plugin-database/pull/622. Let's hope they accept it and we can move forward.

EDIT: the pull request has been approved and is now available on Testing Store Channel.

aqlstar commented:
it worked! thank you very much!

wynn1212 commented:
Hello @popsUlfr,

Since the updated plugin has been approved and merged into the Decky plugin database and is now available in the main store channel, I suggest merging this pull request into your repository to maintain consistency and support future development.

Once again, big thanks to @popsUlfr for approving me to submit the updated plugin to the Decky plugin database.

AkazaRenn committed:
Change the toggle to "auto pause or not"

AkazaRenn committed:
Improve settings for a bit

AkazaRenn committed:
New settings.ts

AkazaRenn committed:
Migrate to patching

AkazaRenn committed:
Bump version to 0.5.0

AkazaRenn committed:
fix large ID conversion

AkazaRenn committed:
Revert version bump

AkazaRenn committed:
Fix the logic for noAutoPauseSet, totally got it opposited

AkazaRenn committed:
Swap pause and play icon since it's a button now

wynn1212 committed:
Merge pull request #1 from AkazaRenn/main

Several changes

AkazaRenn committed:
Squashed commit of the following:

commit 1c8afb587583417ce3fc7dfc0bff3d8e59a72e61
Author: Akaza Renn <8340896+AkazaRenn@users.noreply.github.com>
Date:   Sun Jul 7 18:39:12 2024 -0400

    Update play/pause icon color

commit 730f8bf4ceba56272dacf4539563dc559cc212f0
Author: Akaza Renn <8340896+AkazaRenn@users.noreply.github.com>
Date:   Sun Jul 7 18:17:06 2024 -0400

    Update template to latest main

commit c5996fc10c6b1c9eded2b0a174df38721289170d
Author: Akaza Renn <8340896+AkazaRenn@users.noreply.github.com>
Date:   Sun Jul 7 18:01:06 2024 -0400

    Update document

commit 7a0e0d597fe106e2d20acc3ffa53a1a7f42bf23a
Author: Akaza Renn <8340896+AkazaRenn@users.noreply.github.com>
Date:   Sun Jul 7 17:57:15 2024 -0400

    Adjust style

commit 46e84ab535ad69e45e5fa85ef590e0ffa4300ce7
Author: Akaza Renn <8340896+AkazaRenn@users.noreply.github.com>
Date:   Sun Jul 7 17:30:29 2024 -0400

    Button over icon

AkazaRenn committed:
Fix pausing click and games without an icon

AkazaRenn committed:
Update getAppIcon function

AkazaRenn committed:
Remove the zip - shouldn't be here

AkazaRenn committed:
Change sticky behavior to always flip

wynn1212 committed:
Merge pull request #2 from AkazaRenn/main

Document update, merge pause button with game icon

AkazaRenn committed:
Code clean up

wynn1212 committed:
Update documents and screenshots.

wynn1212 committed:
Update documents and screenshots.

AkazaRenn committed:
update dependencies

AkazaRenn committed:
Squashed commit of the following:

commit 9288faf1f8f854cdb0fce4f7d3794f99b38eb883
Author: Tseng Wynn <wynn1212@gmail.com>
Date:   Tue Jul 9 00:34:52 2024 +0800

    Update documents and screenshots.

commit 4b3ea846f85fcb26828b3aa556370535650365de
Author: Tseng Wynn <wynn1212@gmail.com>
Date:   Tue Jul 9 00:34:52 2024 +0800

    Update documents and screenshots.

commit 147447acdb5644d01fb4694d4d23845842a9652d
Merge: 4ee9700 d841396
Author: Tseng Wynn <wynn1212@gmail.com>
Date:   Mon Jul 8 22:38:48 2024 +0800

    Merge pull request #2 from AkazaRenn/main

    Document update, merge pause button with game icon

commit 4ee97008798c2847255d65e9083f83bed0c27a22
Merge: e4b510f b8a35d8
Author: Tseng Wynn <wynn1212@gmail.com>
Date:   Mon Jul 8 01:56:46 2024 +0800

    Merge pull request #1 from AkazaRenn/main

    Several changes

AkazaRenn committed:
Unfinished: left/right navigation

AkazaRenn committed:
Limit the width of marquee

AkazaRenn committed:
Get rid of the hard coded focusable width

AkazaRenn committed:
Reorder imports

AkazaRenn committed:
Dependency update

AkazaRenn committed:
Match the paddings of Marquee

wynn1212 committed:
Merge pull request #3 from AkazaRenn/main

Left/Right navigation between pause button and toggle

AkazaRenn committed:
Minor adjusts

AkazaRenn committed:
Add my fediverse info to package.json

wynn1212 commented:
Hi @popsUlfr,

Given your current situation, and with the recent fixes made by @AkazaRenn and myself, could you approve us to take over and continue this project?

AkazaRenn committed:
Hide autoPauseSet toggle when autoPause is false

AkazaRenn committed:
Actions to build plugin on each main push

AkazaRenn committed:
Update actino to do a release

AkazaRenn committed:
Squashed commit of the following:

commit f5c9f449cb0652e5ca8d18a589959abdbf94b323
Author: Akaza Renn <8340896+AkazaRenn@users.noreply.github.com>
Date:   Tue Sep 24 14:29:22 2024 -0400

    Add pakcageManager entry to package.json

commit a050768780fd0fe6575538f5b52b5c85cd615fae
Merge: d80ef41 d69496d
Author: Akaza Renn <8340896+AkazaRenn@users.noreply.github.com>
Date:   Tue Sep 24 14:25:25 2024 -0400

    Merge remote-tracking branch 'template/main' into dev

commit d69496d4adc30597bef8fdec377fd280ac9f1a00
Author: Wayne Heaney <42350981+wheaney@users.noreply.github.com>
Date:   Wed Jul 24 15:01:55 2024 -0700

    Add _uninstall function to the python Plugin template (#34)

commit a85c79735f2ff204325b5e930dad1369439d2db1
Author: TrainDoctor <11465594+TrainDoctor@users.noreply.github.com>
Date:   Wed Jul 24 15:00:07 2024 -0700

    Update README.md

    Closes #42

commit d80ef41e91537d12321ea9736ab21d8fc7136e10
Author: Akaza Renn <8340896+AkazaRenn@users.noreply.github.com>
Date:   Sun Jul 21 16:12:55 2024 -0400

    Update decky library

commit 243ff14613067e4a8f41de3bc87a1b78ca107865
Merge: 69b3ac6 2840dc0
Author: Akaza Renn <8340896+AkazaRenn@users.noreply.github.com>
Date:   Sun Jul 21 16:07:38 2024 -0400

    Merge branch 'decky-v3.0.0+' into dev

commit 69b3ac618a7fbdf425f60d67183a0d2f0272597c
Merge: ff4bd85 93ef110
Author: Akaza Renn <8340896+AkazaRenn@users.noreply.github.com>
Date:   Sun Jul 21 16:05:28 2024 -0400

    Merge remote-tracking branch 'template/aa/websockets' into dev

commit ff4bd85b6277b64ecf205e09046ec862b7ff3012
Author: Akaza Renn <8340896+AkazaRenn@users.noreply.github.com>
Date:   Sun Jul 21 16:03:35 2024 -0400

    new line on end of setup.sh

commit 08de19500423f2970179466264d14715ba0a12bc
Merge: d406b78 d18cbd5
Author: Akaza Renn <8340896+AkazaRenn@users.noreply.github.com>
Date:   Sun Jul 21 16:01:29 2024 -0400

    Merge remote-tracking branch 'template/main' into dev

commit 2840dc0bc9c364cb02107e40c02d0e53794662de
Merge: f97cfa5 9f06313
Author: Akaza Renn <8340896+AkazaRenn@users.noreply.github.com>
Date:   Sat Jul 13 21:37:15 2024 -0400

    Merge branch 'main' into decky-v3.0.0+

commit d18cbd5c992cf6dbe52bd83942315139743ce753
Author: Renn <8340896+AkazaRenn@users.noreply.github.com>
Date:   Wed Jul 10 00:59:14 2024 -0400

    Fix deployment for plugin names with more than one spaces (#41)

commit f97cfa5c145d79a6baa626392972a44265f695ec
Author: Akaza Renn <8340896+AkazaRenn@users.noreply.github.com>
Date:   Mon Jul 8 21:25:01 2024 -0400

    Log error in loadAll()

commit 39def3e70427eb892334bae8428d76a1d7b8901a
Author: Akaza Renn <8340896+AkazaRenn@users.noreply.github.com>
Date:   Mon Jul 8 21:17:26 2024 -0400

    Reverse some changes that don't belong here

commit 748a75fd8c39abf6eec870f4b28f6045d951a9f8
Author: Akaza Renn <8340896+AkazaRenn@users.noreply.github.com>
Date:   Mon Jul 8 21:11:14 2024 -0400

    Squashed commit of the following:

    commit 9abd43e6b10aadab81e23bef2896147548dd4b68
    Author: Akaza Renn <8340896+AkazaRenn@users.noreply.github.com>
    Date:   Mon Jul 8 21:07:03 2024 -0400

        Test build

    commit 289fb70cb07b0e2ce0a91ba643869698e56e4c98
    Merge: 5a8463a 9288faf
    Author: Akaza Renn <8340896+AkazaRenn@users.noreply.github.com>
    Date:   Mon Jul 8 21:05:59 2024 -0400

        Merge remote-tracking branch 'upstream/main' into newapi

    commit 5a8463a066c2427a456f2f41a49a6678519907e5
    Merge: a613571 93ef110
    Author: Akaza Renn <8340896+AkazaRenn@users.noreply.github.com>
    Date:   Mon Jul 8 21:04:12 2024 -0400

        Merge remote-tracking branch 'template/aa/websockets' into newapi

    commit a613571b46526030db28899ad0acb4d3aad6f264
    Author: Akaza Renn <8340896+AkazaRenn@users.noreply.github.com>
    Date:   Mon Jul 8 20:47:27 2024 -0400

        Merge branch 'main' into newapi

    commit 93ef11096f2a0a094de2b7094b817f110bc5ea3b
    Author: AAGaming <aagaming@riseup.net>
    Date:   Mon Jul 8 20:28:57 2024 -0400

        use @decky/rollup rollup config template

    commit 9288faf1f8f854cdb0fce4f7d3794f99b38eb883
    Author: Tseng Wynn <wynn1212@gmail.com>
    Date:   Tue Jul 9 00:34:52 2024 +0800

        Update documents and screenshots.

    commit 4b3ea846f85fcb26828b3aa556370535650365de
    Author: Tseng Wynn <wynn1212@gmail.com>
    Date:   Tue Jul 9 00:34:52 2024 +0800

        Update documents and screenshots.

    commit 147447acdb5644d01fb4694d4d23845842a9652d
    Merge: 4ee9700 d841396
    Author: Tseng Wynn <wynn1212@gmail.com>
    Date:   Mon Jul 8 22:38:48 2024 +0800

        Merge pull request #2 from AkazaRenn/main

        Document update, merge pause button with game icon

    commit 603afe413b9d67b211fe915fe53579d453a1bd41
    Author: Akaza Renn <8340896+AkazaRenn@users.noreply.github.com>
    Date:   Sun Jul 7 16:06:22 2024 -0400

        Remove usage of VFC

    commit 4ee97008798c2847255d65e9083f83bed0c27a22
    Merge: e4b510f b8a35d8
    Author: Tseng Wynn <wynn1212@gmail.com>
    Date:   Mon Jul 8 01:56:46 2024 +0800

        Merge pull request #1 from AkazaRenn/main

        Several changes

    commit 3eb9e950494e3e7e7800abaec6df0944a1d59f65
    Author: Akaza Renn <8340896+AkazaRenn@users.noreply.github.com>
    Date:   Sat Jul 6 15:16:40 2024 -0400

        Move interop to another file

    commit efe4b4917d4533f989a299854a7cae3e0b13287e
    Author: Akaza Renn <8340896+AkazaRenn@users.noreply.github.com>
    Date:   Sat Jul 6 12:08:36 2024 -0400

        support new APIs

    commit d601b1d7c545ed053a10a77373d7007d15f2db90
    Author: Akaza Renn <8340896+AkazaRenn@users.noreply.github.com>
    Date:   Sat Jul 6 10:48:28 2024 -0400

        Update description

    commit 64310bcf71c5e9ec463a840642968900e2f4882d
    Author: AAGaming <aagaming@riseup.net>
    Date:   Thu Jun 13 18:28:56 2024 -0400

        update deps

    commit b939274d157257d0a57b339193d961b39ded6207
    Author: AAGaming <aagaming@riseup.net>
    Date:   Thu Jun 13 18:11:44 2024 -0400

        support new APIs fully

    commit cbd489150fc2f721fe788ef6d58b8b792843e7d0
    Author: AnOpenSauceDev <amongus17562@gmail.com>
    Date:   Tue Feb 27 04:03:18 2024 +1000

        Make `setup.sh` download the correct binaries. (#37)

        `setup.sh` downloads non-existent binaries for Decky CLI, creating a file with the contents "Not Found". This puts a check for both OS and arch so the right files are downloaded.

    commit 4ab713585dbe77baefa83c3a2fa57cb1872be1f4
    Author: TrainDoctor <11465594+TrainDoctor@users.noreply.github.com>
    Date:   Mon Feb 12 18:12:48 2024 -0800

        Update LICENSE

    commit 82cb4860e0b069f6feb259f7d68d78a65896dacc
    Author: TrainDoctor <traindoctor@protonmail.com>
    Date:   Sun Feb 4 18:51:12 2024 -0800

        Update DFL and fix navigation

    commit 498ca9b13d5fc951a4d1be7b0f4e0f5e9aef0fc6
    Author: TrainDoctor <traindoctor@protonmail.com>
    Date:   Sun Feb 4 17:51:53 2024 -0800

        Support spaces in pwd

    commit 8f6c61207f4afe623d9f00b2cca3681434d9d4de
    Author: AAGaming <aagaming@riseup.net>
    Date:   Mon Oct 16 19:30:50 2023 -0400

        fix: add jsxFragmentFactory to tsconfig

    commit a193ffad524c4c1ce66b17f461410cf041dec375
    Author: jurassicplayer <jurassicplayer75@gmail.com>
    Date:   Fri Sep 8 14:33:52 2023 -0700

        Update defsettings.json (#32)

        Set default to the default hostname to possibly remove setup effort

    commit 5a91054ba8d10d72b7838ae3677670f05a5757d3
    Author: Jari <safijari@users.noreply.github.com>
    Date:   Thu Aug 24 19:30:34 2023 -0500

        pass serverAPI correctly to Content (#29)

    commit ec1dfc791c3c9c6fa26e51b3997a0a0da9f090a5
    Author: TrainDoctor <11465594+TrainDoctor@users.noreply.github.com>
    Date:   Thu Aug 3 17:27:35 2023 -0700

        Remove spaces properly

        Remove all spaces and replace with dashes as originally intended.

    commit ed314a5128235972cf8adeaf29928deaa9fdd77f
    Author: TrainDoctor <11465594+TrainDoctor@users.noreply.github.com>
    Date:   Sun Jun 18 17:54:08 2023 -0700

        Update README.md

    commit 3d64a3a076c5d29389178cbfeaa279dd279e4980
    Author: TrainDoctor <11465594+TrainDoctor@users.noreply.github.com>
    Date:   Sat Jun 10 21:26:45 2023 -0700

        Update LICENSE

    commit 532e0b8a8789b3636236db5e67ef61effaa5bb29
    Author: TrainDoctor <11465594+TrainDoctor@users.noreply.github.com>
    Date:   Wed Jun 7 11:59:23 2023 -0700

        Update setup.sh

    commit 625eaf2897c95267fa0fb795103861225fe12065
    Author: Yao Chung Hu <30311066+FlashyReese@users.noreply.github.com>
    Date:   Mon Jun 5 03:40:58 2023 +0800

        Fix setup cli detection (#26)

    commit 4beeece5127a149efbd29394f09d68b04da4f9e8
    Author: Yao Chung Hu <30311066+FlashyReese@users.noreply.github.com>
    Date:   Mon Jun 5 02:36:10 2023 +0800

        Global DFL Part 2 (#25)

    commit e0ec175d16156f50b87374c14d8c9ee9f2c613a0
    Author: TrainDoctor <11465594+TrainDoctor@users.noreply.github.com>
    Date:   Sat Jun 3 07:42:27 2023 -0700

        Global dfl (#24)

        * Initial global DFL support

        * Update DFL

        * Remove unmaintained makefile

        * Move to CLI

        * Properly extract zip files!

        ---------

        Co-authored-by: AAGaming <aa@mail.catvibers.me>

    commit ecb45e71f1123f13ef29c06b71ee6c0a4e193e19
    Author: TrainDoctor <traindoctor@protonmail.com>
    Date:   Sun May 14 17:45:34 2023 -0700

        DFL Bump

    commit cadd0bee9e3f2df1b788bb0e6353bab1af1455e2
    Author: TrainDoctor <traindoctor@protonmail.com>
    Date:   Fri Apr 14 18:32:33 2023 -0700

        Update DFL

    commit daec6099c439e76d01b89215e7bdebce615d0cd6
    Author: Matt Carlucci <mcarlucci@users.noreply.github.com>
    Date:   Wed Apr 12 16:45:41 2023 -0500

        updated lockfileVersion to 6.0, updated README (#23)

    commit aeda6de1d6cbc04bc2e062540d4ff1051d2d3967
    Author: Jari <safijari@users.noreply.github.com>
    Date:   Wed Mar 15 21:00:31 2023 -0500

        Bump DFL to latest (#22)

    commit b0bf2b1cd1424b174c1c449c34711a48a2c1115e
    Author: TrainDoctor <traindoctor@protonmail.com>
    Date:   Sun Mar 5 19:03:54 2023 -0800

        Revert due to changes breaking plugins

    commit 6df09f9fcb1d8331bacab52e3c9266eeb8f76f3f
    Author: TrainDoctor <traindoctor@protonmail.com>
    Date:   Sun Mar 5 16:39:28 2023 -0800

        Added USER_ID and DECKY_USER_ID env vars

    commit 16f44fc8b127b68a864678115c7e165a51691f63
    Author: kmiqka <6089748+kmiqka@users.noreply.github.com>
    Date:   Sat Mar 4 03:47:26 2023 +0100

        corrects missing ssh port for rsync (#20)

    commit 1dd9bb21b6571c878d29f48372ca02f0471dca49
    Author: Kevin Hester <kevinh@geeksville.com>
    Date:   Mon Feb 20 10:34:55 2023 -0800

        Add a 'restartloader' tasks to restart the python backend. (#19)

        This is useful for testing/developing the python code.

        I also added this as an optional last
        subtask on allinone.

    commit d81b0bbc060eca5276763fe25c8010dab82b8dad
    Author: Aleksandr <xndr78@users.noreply.github.com>
    Date:   Mon Feb 20 00:46:54 2023 +0200

        Makefile helper tools, bump frontend UI version (#17)

        * Makefile helper tools, frontend UI lib updated, option added to build plugin in docker

        * docker/makefile cleanup

    commit 39f7aadf70ce9da779d3138183ce82ba10c0fd53
    Author: Philipp Richter <richterphilipp.pops@gmail.com>
    Date:   Sun Feb 19 22:43:09 2023 +0000

        Add support for the 'decky_plugin' module exposed by decky-loader (#16)

        * Add support for the 'decky_plugin' module exposed by decky-loader

        * Make versioned 'decky_plugin.pyi' stub interface and 'py_modules'
        support available

        * Add rule to ignore 'decky_plugin.py'

    commit b05ecd55e7d049e0951b53fbe4aa2f329f9db905
    Author: TrainDoctor <traindoctor@protonmail.com>
    Date:   Thu Feb 9 10:09:33 2023 -0800

        Update README.md

    commit 4991e260a4157b033ce8487fbc8501d4a6151817
    Author: TrainDoctor <traindoctor@protonmail.com>
    Date:   Wed Feb 8 18:44:07 2023 -0800

        Update README.md

    commit ae179d618e0c02c57718d351089666cea801ed93
    Author: Jens Klingenberg <mail@jensklingenberg.de>
    Date:   Tue Jan 10 00:56:13 2023 +0100

        Replace NavigateToStore (#15)

    commit 9b0375493972630213b3d61ac119d5bc4a5dcb8f
    Author: Luis Finke <luisfinke@gmail.com>
    Date:   Fri Jan 6 17:36:19 2023 -0500

        add py_modules to vscode defsettings.json (#14)

    commit 965b1ea7d021e0e0e574191b2c4e0f15be8d6363
    Author: Jens Klingenberg <mail@jensklingenberg.de>
    Date:   Thu Jan 5 01:02:31 2023 +0100

        fix broken link (#13)

    commit e2b4542b58f634998db62b9b69aeeac6130e5ea5
    Author: TrainDoctor <traindoctor@protonmail.com>
    Date:   Sat Nov 19 13:36:20 2022 -0800

        Update README.md

    commit 36389449ff2be245c03f2302d0c74e382663705c
    Author: TrainDoctor <traindoctor@protonmail.com>
    Date:   Sat Nov 5 15:43:10 2022 -0700

        Update README.md

    commit 17924b84e32e5576d5e7e55b86d8279881a36b06
    Author: TrainDoctor <traindoctor@protonmail.com>
    Date:   Sat Nov 5 03:42:56 2022 -0700

        Explicit requirements for LICENSE.md

    commit 9a58f8aaa0aec88857cd07603390a96871b4dabe
    Author: TrainDoctor <traindoctor@protonmail.com>
    Date:   Sat Oct 22 18:53:22 2022 -0700

        Update template

    commit 05eefd80a5315c6cd7e72718878682d7de29f56f
    Author: TrainDoctor <traindoctor@protonmail.com>
    Date:   Sat Oct 22 18:48:59 2022 -0700

        Update main.py

    commit 8dd0b92dc09906e19cc19a22222ca06e74914963
    Author: TrainDoctor <traindoctor@protonmail.com>
    Date:   Sun Oct 9 16:17:35 2022 -0700

        Update LICENSE

    commit 1228d24ccc4457d5678fdf69a563e09c3282ce84
    Author: TrainDoctor <traindoctor@protonmail.com>
    Date:   Sun Oct 9 08:55:03 2022 -0700

        Fix plugin template having the do not upload tag

    commit fee981b521e77c40b80e9abcf0f37cc64b553906
    Author: TrainDoctor <traindoctor@protonmail.com>
    Date:   Mon Sep 19 14:29:37 2022 -0700

        Update package.json

    commit 33bde6f55f4ff4709053cbcd9fce286a022deca0
    Author: TrainDoctor <traindoctor@protonmail.com>
    Date:   Mon Sep 19 14:29:13 2022 -0700

        Update package.json

    commit b0f960d7db4f8363294de6f5ef79207273eb3af1
    Author: TrainDoctor <traindoctor@protonmail.com>
    Date:   Sun Sep 18 17:21:42 2022 -0700

        Update package.json

    commit 55d4daca69ba9122a137f58c60151953b853b36c
    Author: TrainDoctor <traindoctor@protonmail.com>
    Date:   Wed Aug 10 14:11:27 2022 -0700

        Added "do not upload" tag to plugin-template

    commit c15ae3b7438ba880f3acc3774d4e5e41eba4c436
    Author: TrainDoctor <traindoctor@protonmail.com>
    Date:   Wed Aug 10 13:33:24 2022 -0700

        Big README update and comments in Dockerfile

    commit 7517e7e1c7edf59cbf0c64898bcccd06865dccc5
    Author: TrainDoctor <traindoctor@protonmail.com>
    Date:   Tue Aug 9 20:27:47 2022 -0700

        Made entrypoint.sh executable

    commit 43d9b29d04199588058debe470ec0f0a44125e0d
    Author: TrainDoctor <traindoctor@protonmail.com>
    Date:   Tue Aug 9 20:20:31 2022 -0700

        Adjusted entrypoint.sh to actually build binary

    commit aa46a9ee70415c45214d101b723765f1ba2ebbe8
    Author: TrainDoctor <traindoctor@protonmail.com>
    Date:   Tue Aug 9 19:41:05 2022 -0700

        Updated dockerfile to reflect CI/CD on database

    commit 9d8ffe8299e7a6db0c31dfeb4ca4b57f1022cf36
    Author: TrainDoctor <traindoctor@protonmail.com>
    Date:   Tue Aug 9 19:17:47 2022 -0700

        Add default entrypoint.sh

    commit 3f233cae29e2b59eeefe90886531f6490ec651a2
    Author: TrainDoctor <traindoctor@protonmail.com>
    Date:   Mon Aug 8 14:15:25 2022 -0700

        Switch to holo-docker image

    commit d13987ac2be6232d533f9d973150f02b7c7a5912
    Author: TrainDoctor <traindoctor@protonmail.com>
    Date:   Wed Jul 27 18:17:05 2022 -0700

        whoops wrong image name

    commit 328d3a83162832ecc21720ee031481d853b5c621
    Author: TrainDoctor <traindoctor@protonmail.com>
    Date:   Wed Jul 27 16:52:49 2022 -0700

        Add template dockerfile for testing

    commit 95ee2b54c723013530992beb005ac16e6898975d
    Author: TrainDoctor <traindoctor@protonmail.com>
    Date:   Sun Sep 11 16:41:34 2022 -0700

        Add remote_binary, use url to download remote bin

    commit eef6195ec47d58fdfe9cb1a3bda870dcefd3cb54
    Author: TrainDoctor <traindoctor@protonmail.com>
    Date:   Sat Sep 10 19:10:02 2022 -0700

        Update decky-frontend-lib

    commit 4fcd72fd1b17ce9df4db279507ccd47328f03925
    Author: TrainDoctor <traindoctor@protonmail.com>
    Date:   Wed Aug 24 18:06:26 2022 -0700

        Added basic logging config

    commit 2c0988f3c111d078608450048b409f367210eb45
    Author: TrainDoctor <traindoctor@protonmail.com>
    Date:   Tue Aug 23 11:06:04 2022 -0700

        Update README.md

    commit 7fcb7739085512e8ba1e85579956d5b67d1c361e
    Author: TrainDoctor <traindoctor@protonmail.com>
    Date:   Tue Aug 23 11:05:20 2022 -0700

        Update README.md

    commit 7972ff7879c01613e88bfb09283217f8c124af7d
    Author: Nick Girga <50096660+nickgirga@users.noreply.github.com>
    Date:   Sun Aug 21 10:38:27 2022 -0700

        fix link typo in README.md (#8)

        Co-authored-by: Nick Girga <contact@nickgirga.com>

    commit 5b4aa87ba310ef96239977f815aa883e844f2bbe
    Author: TrainDoctor <traindoctor@protonmail.com>
    Date:   Mon Aug 15 15:00:00 2022 -0700

        Add a license to the template

        You may re-license your plugin under another license but you will need to keep this license notice with source code and binary redistribution.
        This is best done by adding your preferred license to the top of this file and including this license above.

    commit af817b6b5f530a1a062cee708043dc3a41155aba
    Author: TrainDoctor <traindoctor@protonmail.com>
    Date:   Sat Aug 13 20:14:43 2022 -0700

        Merge Dockerfile and example files for backend support example (#7)

        * Add template dockerfile for testing

        * Switch to holo-docker image

        * Add default entrypoint.sh

        * Updated dockerfile to reflect CI/CD on database

        * Add template makefile, main.c

        * Adjusted entrypoint.sh to actually build binary

        * Made entrypoint.sh executable

        * Big README update and comments in Dockerfile

    commit 41103641bd319468a06558ca97aef596a78d9f34
    Author: TrainDoctor <traindoctor@protonmail.com>
    Date:   Thu Jul 28 15:16:49 2022 -0700

        Update README.md

    commit 9502211299861729d3b2250b60aae75cf8b2700e
    Author: TrainDoctor <traindoctor@protonmail.com>
    Date:   Thu Jul 28 15:13:20 2022 -0700

        Update README.md

    commit 3c508eb258e7728116afd4b0c8cbe99008eecdae
    Author: D_Pain <dpainhahn@gmail.com>
    Date:   Wed Jul 27 13:14:38 2022 -0700

        Ensure correct permissions during deployment. (#6)

        * Sets files and directories that are being deployed to the Steamdeck to have correct permissions.

    commit b88b430e2696453cb518eb7d4b4e615566114eaf
    Author: TrainDoctor <traindoctor@protonmail.com>
    Date:   Tue Jul 26 13:39:05 2022 -0700

        Update to latest version of decky-frontend-library

    commit 93c909b231fbaa57304a2346933440bde092ef32
    Author: TrainDoctor <traindoctor@protonmail.com>
    Date:   Tue Jul 26 13:36:49 2022 -0700

        Missed a spot, this will make all subdirs of homebrew readwrite-able
        Keep in mind this does not reflect the default setup for end-users!

    commit ac0fde6d6e90f60b4f09bd4cb660581bf194652f
    Author: TrainDoctor <traindoctor@protonmail.com>
    Date:   Mon Jul 25 12:12:38 2022 -0700

        Update tasks to deploy plugins to non-dev directory
        Now that binary releases are out, this makes more sense for developers.

    commit 6639643077cbc49007afc215c7cb3a5e6a866fb0
    Author: TrainDoctor <traindoctor@protonmail.com>
    Date:   Mon Jul 18 16:15:20 2022 -0700

        Add defaults folder to template

    commit da3a4183cabff6bdc382e3946ddb45ef252dbd0b
    Author: AAGaming <aa@mail.catvibers.me>
    Date:   Sun Jul 3 17:54:05 2022 -0400

        bump decky-frontend-lib

    commit d58b8f752d3b1502cdae0ab9c646516c2838db4a
    Author: TrainDoctor <11465594+TrainDoctor@users.noreply.github.com>
    Date:   Wed Jun 29 12:19:09 2022 -0700

        Run chmodfolders before and after deploy

    commit 25f3f8c7e19780f479923bef633146274d0e1bdf
    Author: AAGaming <aa@mail.catvibers.me>
    Date:   Tue Jun 28 14:10:09 2022 -0400

        another temp edit

    commit bc8b57be3af07022c1b6fab07580ee6c22406df2
    Author: AAGaming <aa@mail.catvibers.me>
    Date:   Tue Jun 28 13:45:27 2022 -0400

        temp plugin.json fixes for store testing

    commit e3e6f23c7326062846ef95653641a270d922d74e
    Author: TrainDoctor <11465594+TrainDoctor@users.noreply.github.com>
    Date:   Thu Jun 16 15:04:18 2022 -0700

        Remove updatefrontendlib, will only be used as needed

    commit a40d0da7bb3a1cbca83547a59177a007b72340de
    Author: TrainDoctor <11465594+TrainDoctor@users.noreply.github.com>
    Date:   Thu Jun 16 14:59:43 2022 -0700

        They were, in fact... doing something

    commit 604b355cceb68e8d746a43c36c904b93c7f3b212
    Author: TrainDoctor <11465594+TrainDoctor@users.noreply.github.com>
    Date:   Thu Jun 16 12:47:53 2022 -0700

        Chmod task to fix perms

    commit f240cbc23f8103fc280a41a15494509529c90b9c
    Author: TrainDoctor <11465594+TrainDoctor@users.noreply.github.com>
    Date:   Thu Jun 16 12:15:06 2022 -0700

        Working deploy task to send to deck

    commit b0d5050896c5ba7a597f82e675c3fc68185eb2cd
    Author: TrainDoctor <11465594+TrainDoctor@users.noreply.github.com>
    Date:   Tue Jun 14 15:01:10 2022 -0700

        This is broken but new job awaits

    commit d67d18ab7766e8dbb79fd48ac347d601f82d3332
    Author: TrainDoctor <11465594+TrainDoctor@users.noreply.github.com>
    Date:   Tue Jun 14 13:38:29 2022 -0700

        Add README, tasks, moved to pnpm and added an ignore for react (#3)

    commit 26dc961f583f66b8333d4f4ddbdba51c00676a7e
    Author: AAGaming <aa@bigdumb.gq>
    Date:   Wed Jun 1 17:46:56 2022 -0400

        showModal->showContextMenu

    commit d8870f20620fc459ed284f652eb204366e6e317d
    Author: Jonas Dellinger <jonas@dellinger.dev>
    Date:   Mon May 30 22:23:13 2022 +0200

        Update lib, show example of router

    commit 82320806496a456a67bd87a844e278c535269b75
    Author: AAGaming <aa@bigdumb.gq>
    Date:   Sat May 28 23:19:40 2022 -0400

        fix oops

    commit 6e73b2796137984e5290a5d5f3e381362f21bfea
    Merge: adb8a84 a5d4912
    Author: AAGaming <aa@bigdumb.gq>
    Date:   Sat May 28 21:52:51 2022 -0400

        Merge branch 'main' of github.com:SteamDeckHomebrew/decky-plugin-template

    commit adb8a84d2c313bedd6d6e46408056c3a2677ddb3
    Author: AAGaming <aa@bigdumb.gq>
    Date:   Sat May 28 21:50:47 2022 -0400

        fix react

    commit e264d9754cc0ebbdd8f49994872a85031459d436
    Author: AAGaming <aa@bigdumb.gq>
    Date:   Sat May 28 21:50:24 2022 -0400

        actually fix react

    commit a5d491256fd058796162ef9e234b7b5fea6b6ef0
    Author: Jonas Dellinger <jonas@dellinger.dev>
    Date:   Thu May 26 13:33:25 2022 +0200

        Use new exported components

    commit f554798ee4b92e9e4e7086485fda8ad49ea4bb61
    Author: Jonas Dellinger <jonas@dellinger.dev>
    Date:   Wed May 25 21:34:31 2022 +0200

        Added support for static assets

    commit 76470f49a82ecd8b32344aaf8b0186676d948496
    Author: AAGaming <aa@bigdumb.gq>
    Date:   Fri May 20 23:50:25 2022 -0400

        readd npm lockfile

    commit 569f8b8707636cb52d9b89278c28150d668175db
    Author: AAGaming <aa@bigdumb.gq>
    Date:   Fri May 20 18:13:31 2022 -0400

        fix dependencies, edit rollup and ts config, switch jsx runtime, use pnpm lockfiles

    commit 5a6001026860aca724bf5aa871da93a5edabe12c
    Author: Jonas Dellinger <jonas@dellinger.dev>
    Date:   Fri May 13 23:18:19 2022 +0200

        Actuall add decky-frontend-lib and remove some old code

    commit 52f6ca1c587df2cc74b986bc82419c867c404f47
    Author: Jonas Dellinger <jonas@dellinger.dev>
    Date:   Sat Apr 23 00:42:11 2022 +0200

        Initial commit

commit 93ef11096f2a0a094de2b7094b817f110bc5ea3b
Author: AAGaming <aagaming@riseup.net>
Date:   Mon Jul 8 20:28:57 2024 -0400

    use @decky/rollup rollup config template

commit 64310bcf71c5e9ec463a840642968900e2f4882d
Author: AAGaming <aagaming@riseup.net>
Date:   Thu Jun 13 18:28:56 2024 -0400

    update deps

commit b939274d157257d0a57b339193d961b39ded6207
Author: AAGaming <aagaming@riseup.net>
Date:   Thu Jun 13 18:11:44 2024 -0400

    support new APIs fully

commit cbd489150fc2f721fe788ef6d58b8b792843e7d0
Author: AnOpenSauceDev <amongus17562@gmail.com>
Date:   Tue Feb 27 04:03:18 2024 +1000

    Make `setup.sh` download the correct binaries. (#37)

    `setup.sh` downloads non-existent binaries for Decky CLI, creating a file with the contents "Not Found". This puts a check for both OS and arch so the right files are downloaded.

commit 4ab713585dbe77baefa83c3a2fa57cb1872be1f4
Author: TrainDoctor <11465594+TrainDoctor@users.noreply.github.com>
Date:   Mon Feb 12 18:12:48 2024 -0800

    Update LICENSE

commit 82cb4860e0b069f6feb259f7d68d78a65896dacc
Author: TrainDoctor <traindoctor@protonmail.com>
Date:   Sun Feb 4 18:51:12 2024 -0800

    Update DFL and fix navigation

commit 498ca9b13d5fc951a4d1be7b0f4e0f5e9aef0fc6
Author: TrainDoctor <traindoctor@protonmail.com>
Date:   Sun Feb 4 17:51:53 2024 -0800

    Support spaces in pwd

commit 8f6c61207f4afe623d9f00b2cca3681434d9d4de
Author: AAGaming <aagaming@riseup.net>
Date:   Mon Oct 16 19:30:50 2023 -0400

    fix: add jsxFragmentFactory to tsconfig

commit a193ffad524c4c1ce66b17f461410cf041dec375
Author: jurassicplayer <jurassicplayer75@gmail.com>
Date:   Fri Sep 8 14:33:52 2023 -0700

    Update defsettings.json (#32)

    Set default to the default hostname to possibly remove setup effort

commit 5a91054ba8d10d72b7838ae3677670f05a5757d3
Author: Jari <safijari@users.noreply.github.com>
Date:   Thu Aug 24 19:30:34 2023 -0500

    pass serverAPI correctly to Content (#29)

commit ec1dfc791c3c9c6fa26e51b3997a0a0da9f090a5
Author: TrainDoctor <11465594+TrainDoctor@users.noreply.github.com>
Date:   Thu Aug 3 17:27:35 2023 -0700

    Remove spaces properly

    Remove all spaces and replace with dashes as originally intended.

commit ed314a5128235972cf8adeaf29928deaa9fdd77f
Author: TrainDoctor <11465594+TrainDoctor@users.noreply.github.com>
Date:   Sun Jun 18 17:54:08 2023 -0700

    Update README.md

commit 3d64a3a076c5d29389178cbfeaa279dd279e4980
Author: TrainDoctor <11465594+TrainDoctor@users.noreply.github.com>
Date:   Sat Jun 10 21:26:45 2023 -0700

    Update LICENSE

commit 532e0b8a8789b3636236db5e67ef61effaa5bb29
Author: TrainDoctor <11465594+TrainDoctor@users.noreply.github.com>
Date:   Wed Jun 7 11:59:23 2023 -0700

    Update setup.sh

commit 625eaf2897c95267fa0fb795103861225fe12065
Author: Yao Chung Hu <30311066+FlashyReese@users.noreply.github.com>
Date:   Mon Jun 5 03:40:58 2023 +0800

    Fix setup cli detection (#26)

commit 4beeece5127a149efbd29394f09d68b04da4f9e8
Author: Yao Chung Hu <30311066+FlashyReese@users.noreply.github.com>
Date:   Mon Jun 5 02:36:10 2023 +0800

    Global DFL Part 2 (#25)

commit e0ec175d16156f50b87374c14d8c9ee9f2c613a0
Author: TrainDoctor <11465594+TrainDoctor@users.noreply.github.com>
Date:   Sat Jun 3 07:42:27 2023 -0700

    Global dfl (#24)

    * Initial global DFL support

    * Update DFL

    * Remove unmaintained makefile

    * Move to CLI

    * Properly extract zip files!

    ---------

    Co-authored-by: AAGaming <aa@mail.catvibers.me>

commit ecb45e71f1123f13ef29c06b71ee6c0a4e193e19
Author: TrainDoctor <traindoctor@protonmail.com>
Date:   Sun May 14 17:45:34 2023 -0700

    DFL Bump

commit cadd0bee9e3f2df1b788bb0e6353bab1af1455e2
Author: TrainDoctor <traindoctor@protonmail.com>
Date:   Fri Apr 14 18:32:33 2023 -0700

    Update DFL

commit daec6099c439e76d01b89215e7bdebce615d0cd6
Author: Matt Carlucci <mcarlucci@users.noreply.github.com>
Date:   Wed Apr 12 16:45:41 2023 -0500

    updated lockfileVersion to 6.0, updated README (#23)

commit aeda6de1d6cbc04bc2e062540d4ff1051d2d3967
Author: Jari <safijari@users.noreply.github.com>
Date:   Wed Mar 15 21:00:31 2023 -0500

    Bump DFL to latest (#22)

commit b0bf2b1cd1424b174c1c449c34711a48a2c1115e
Author: TrainDoctor <traindoctor@protonmail.com>
Date:   Sun Mar 5 19:03:54 2023 -0800

    Revert due to changes breaking plugins

commit 6df09f9fcb1d8331bacab52e3c9266eeb8f76f3f
Author: TrainDoctor <traindoctor@protonmail.com>
Date:   Sun Mar 5 16:39:28 2023 -0800

    Added USER_ID and DECKY_USER_ID env vars

commit 16f44fc8b127b68a864678115c7e165a51691f63
Author: kmiqka <6089748+kmiqka@users.noreply.github.com>
Date:   Sat Mar 4 03:47:26 2023 +0100

    corrects missing ssh port for rsync (#20)

commit 1dd9bb21b6571c878d29f48372ca02f0471dca49
Author: Kevin Hester <kevinh@geeksville.com>
Date:   Mon Feb 20 10:34:55 2023 -0800

    Add a 'restartloader' tasks to restart the python backend. (#19)

    This is useful for testing/developing the python code.

    I also added this as an optional last
    subtask on allinone.

commit d81b0bbc060eca5276763fe25c8010dab82b8dad
Author: Aleksandr <xndr78@users.noreply.github.com>
Date:   Mon Feb 20 00:46:54 2023 +0200

    Makefile helper tools, bump frontend UI version (#17)

    * Makefile helper tools, frontend UI lib updated, option added to build plugin in docker

    * docker/makefile cleanup

commit 39f7aadf70ce9da779d3138183ce82ba10c0fd53
Author: Philipp Richter <richterphilipp.pops@gmail.com>
Date:   Sun Feb 19 22:43:09 2023 +0000

    Add support for the 'decky_plugin' module exposed by decky-loader (#16)

    * Add support for the 'decky_plugin' module exposed by decky-loader

    * Make versioned 'decky_plugin.pyi' stub interface and 'py_modules'
    support available

    * Add rule to ignore 'decky_plugin.py'

commit b05ecd55e7d049e0951b53fbe4aa2f329f9db905
Author: TrainDoctor <traindoctor@protonmail.com>
Date:   Thu Feb 9 10:09:33 2023 -0800

    Update README.md

commit 4991e260a4157b033ce8487fbc8501d4a6151817
Author: TrainDoctor <traindoctor@protonmail.com>
Date:   Wed Feb 8 18:44:07 2023 -0800

    Update README.md

commit ae179d618e0c02c57718d351089666cea801ed93
Author: Jens Klingenberg <mail@jensklingenberg.de>
Date:   Tue Jan 10 00:56:13 2023 +0100

    Replace NavigateToStore (#15)

commit 9b0375493972630213b3d61ac119d5bc4a5dcb8f
Author: Luis Finke <luisfinke@gmail.com>
Date:   Fri Jan 6 17:36:19 2023 -0500

    add py_modules to vscode defsettings.json (#14)

commit 965b1ea7d021e0e0e574191b2c4e0f15be8d6363
Author: Jens Klingenberg <mail@jensklingenberg.de>
Date:   Thu Jan 5 01:02:31 2023 +0100

    fix broken link (#13)

commit e2b4542b58f634998db62b9b69aeeac6130e5ea5
Author: TrainDoctor <traindoctor@protonmail.com>
Date:   Sat Nov 19 13:36:20 2022 -0800

    Update README.md

commit 36389449ff2be245c03f2302d0c74e382663705c
Author: TrainDoctor <traindoctor@protonmail.com>
Date:   Sat Nov 5 15:43:10 2022 -0700

    Update README.md

commit 17924b84e32e5576d5e7e55b86d8279881a36b06
Author: TrainDoctor <traindoctor@protonmail.com>
Date:   Sat Nov 5 03:42:56 2022 -0700

    Explicit requirements for LICENSE.md

commit 9a58f8aaa0aec88857cd07603390a96871b4dabe
Author: TrainDoctor <traindoctor@protonmail.com>
Date:   Sat Oct 22 18:53:22 2022 -0700

    Update template

commit 05eefd80a5315c6cd7e72718878682d7de29f56f
Author: TrainDoctor <traindoctor@protonmail.com>
Date:   Sat Oct 22 18:48:59 2022 -0700

    Update main.py

commit 8dd0b92dc09906e19cc19a22222ca06e74914963
Author: TrainDoctor <traindoctor@protonmail.com>
Date:   Sun Oct 9 16:17:35 2022 -0700

    Update LICENSE

commit 1228d24ccc4457d5678fdf69a563e09c3282ce84
Author: TrainDoctor <traindoctor@protonmail.com>
Date:   Sun Oct 9 08:55:03 2022 -0700

    Fix plugin template having the do not upload tag

commit fee981b521e77c40b80e9abcf0f37cc64b553906
Author: TrainDoctor <traindoctor@protonmail.com>
Date:   Mon Sep 19 14:29:37 2022 -0700

    Update package.json

commit 33bde6f55f4ff4709053cbcd9fce286a022deca0
Author: TrainDoctor <traindoctor@protonmail.com>
Date:   Mon Sep 19 14:29:13 2022 -0700

    Update package.json

commit b0f960d7db4f8363294de6f5ef79207273eb3af1
Author: TrainDoctor <traindoctor@protonmail.com>
Date:   Sun Sep 18 17:21:42 2022 -0700

    Update package.json

commit 55d4daca69ba9122a137f58c60151953b853b36c
Author: TrainDoctor <traindoctor@protonmail.com>
Date:   Wed Aug 10 14:11:27 2022 -0700

    Added "do not upload" tag to plugin-template

commit c15ae3b7438ba880f3acc3774d4e5e41eba4c436
Author: TrainDoctor <traindoctor@protonmail.com>
Date:   Wed Aug 10 13:33:24 2022 -0700

    Big README update and comments in Dockerfile

commit 7517e7e1c7edf59cbf0c64898bcccd06865dccc5
Author: TrainDoctor <traindoctor@protonmail.com>
Date:   Tue Aug 9 20:27:47 2022 -0700

    Made entrypoint.sh executable

commit 43d9b29d04199588058debe470ec0f0a44125e0d
Author: TrainDoctor <traindoctor@protonmail.com>
Date:   Tue Aug 9 20:20:31 2022 -0700

    Adjusted entrypoint.sh to actually build binary

commit aa46a9ee70415c45214d101b723765f1ba2ebbe8
Author: TrainDoctor <traindoctor@protonmail.com>
Date:   Tue Aug 9 19:41:05 2022 -0700

    Updated dockerfile to reflect CI/CD on database

commit 9d8ffe8299e7a6db0c31dfeb4ca4b57f1022cf36
Author: TrainDoctor <traindoctor@protonmail.com>
Date:   Tue Aug 9 19:17:47 2022 -0700

    Add default entrypoint.sh

commit 3f233cae29e2b59eeefe90886531f6490ec651a2
Author: TrainDoctor <traindoctor@protonmail.com>
Date:   Mon Aug 8 14:15:25 2022 -0700

    Switch to holo-docker image

commit d13987ac2be6232d533f9d973150f02b7c7a5912
Author: TrainDoctor <traindoctor@protonmail.com>
Date:   Wed Jul 27 18:17:05 2022 -0700

    whoops wrong image name

commit 328d3a83162832ecc21720ee031481d853b5c621
Author: TrainDoctor <traindoctor@protonmail.com>
Date:   Wed Jul 27 16:52:49 2022 -0700

    Add template dockerfile for testing

commit 95ee2b54c723013530992beb005ac16e6898975d
Author: TrainDoctor <traindoctor@protonmail.com>
Date:   Sun Sep 11 16:41:34 2022 -0700

    Add remote_binary, use url to download remote bin

commit eef6195ec47d58fdfe9cb1a3bda870dcefd3cb54
Author: TrainDoctor <traindoctor@protonmail.com>
Date:   Sat Sep 10 19:10:02 2022 -0700

    Update decky-frontend-lib

commit 4fcd72fd1b17ce9df4db279507ccd47328f03925
Author: TrainDoctor <traindoctor@protonmail.com>
Date:   Wed Aug 24 18:06:26 2022 -0700

    Added basic logging config

commit 2c0988f3c111d078608450048b409f367210eb45
Author: TrainDoctor <traindoctor@protonmail.com>
Date:   Tue Aug 23 11:06:04 2022 -0700

    Update README.md

commit 7fcb7739085512e8ba1e85579956d5b67d1c361e
Author: TrainDoctor <traindoctor@protonmail.com>
Date:   Tue Aug 23 11:05:20 2022 -0700

    Update README.md

commit 7972ff7879c01613e88bfb09283217f8c124af7d
Author: Nick Girga <50096660+nickgirga@users.noreply.github.com>
Date:   Sun Aug 21 10:38:27 2022 -0700

    fix link typo in README.md (#8)

    Co-authored-by: Nick Girga <contact@nickgirga.com>

commit 5b4aa87ba310ef96239977f815aa883e844f2bbe
Author: TrainDoctor <traindoctor@protonmail.com>
Date:   Mon Aug 15 15:00:00 2022 -0700

    Add a license to the template

    You may re-license your plugin under another license but you will need to keep this license notice with source code and binary redistribution.
    This is best done by adding your preferred license to the top of this file and including this license above.

commit af817b6b5f530a1a062cee708043dc3a41155aba
Author: TrainDoctor <traindoctor@protonmail.com>
Date:   Sat Aug 13 20:14:43 2022 -0700

    Merge Dockerfile and example files for backend support example (#7)

    * Add template dockerfile for testing

    * Switch to holo-docker image

    * Add default entrypoint.sh

    * Updated dockerfile to reflect CI/CD on database

    * Add template makefile, main.c

    * Adjusted entrypoint.sh to actually build binary

    * Made entrypoint.sh executable

    * Big README update and comments in Dockerfile

commit 41103641bd319468a06558ca97aef596a78d9f34
Author: TrainDoctor <traindoctor@protonmail.com>
Date:   Thu Jul 28 15:16:49 2022 -0700

    Update README.md

commit 9502211299861729d3b2250b60aae75cf8b2700e
Author: TrainDoctor <traindoctor@protonmail.com>
Date:   Thu Jul 28 15:13:20 2022 -0700

    Update README.md

commit 3c508eb258e7728116afd4b0c8cbe99008eecdae
Author: D_Pain <dpainhahn@gmail.com>
Date:   Wed Jul 27 13:14:38 2022 -0700

    Ensure correct permissions during deployment. (#6)

    * Sets files and directories that are being deployed to the Steamdeck to have correct permissions.

commit b88b430e2696453cb518eb7d4b4e615566114eaf
Author: TrainDoctor <traindoctor@protonmail.com>
Date:   Tue Jul 26 13:39:05 2022 -0700

    Update to latest version of decky-frontend-library

commit 93c909b231fbaa57304a2346933440bde092ef32
Author: TrainDoctor <traindoctor@protonmail.com>
Date:   Tue Jul 26 13:36:49 2022 -0700

    Missed a spot, this will make all subdirs of homebrew readwrite-able
    Keep in mind this does not reflect the default setup for end-users!

commit ac0fde6d6e90f60b4f09bd4cb660581bf194652f
Author: TrainDoctor <traindoctor@protonmail.com>
Date:   Mon Jul 25 12:12:38 2022 -0700

    Update tasks to deploy plugins to non-dev directory
    Now that binary releases are out, this makes more sense for developers.

commit 6639643077cbc49007afc215c7cb3a5e6a866fb0
Author: TrainDoctor <traindoctor@protonmail.com>
Date:   Mon Jul 18 16:15:20 2022 -0700

    Add defaults folder to template

commit da3a4183cabff6bdc382e3946ddb45ef252dbd0b
Author: AAGaming <aa@mail.catvibers.me>
Date:   Sun Jul 3 17:54:05 2022 -0400

    bump decky-frontend-lib

commit d58b8f752d3b1502cdae0ab9c646516c2838db4a
Author: TrainDoctor <11465594+TrainDoctor@users.noreply.github.com>
Date:   Wed Jun 29 12:19:09 2022 -0700

    Run chmodfolders before and after deploy

commit 25f3f8c7e19780f479923bef633146274d0e1bdf
Author: AAGaming <aa@mail.catvibers.me>
Date:   Tue Jun 28 14:10:09 2022 -0400

    another temp edit

commit bc8b57be3af07022c1b6fab07580ee6c22406df2
Author: AAGaming <aa@mail.catvibers.me>
Date:   Tue Jun 28 13:45:27 2022 -0400

    temp plugin.json fixes for store testing

commit e3e6f23c7326062846ef95653641a270d922d74e
Author: TrainDoctor <11465594+TrainDoctor@users.noreply.github.com>
Date:   Thu Jun 16 15:04:18 2022 -0700

    Remove updatefrontendlib, will only be used as needed

commit a40d0da7bb3a1cbca83547a59177a007b72340de
Author: TrainDoctor <11465594+TrainDoctor@users.noreply.github.com>
Date:   Thu Jun 16 14:59:43 2022 -0700

    They were, in fact... doing something

commit 604b355cceb68e8d746a43c36c904b93c7f3b212
Author: TrainDoctor <11465594+TrainDoctor@users.noreply.github.com>
Date:   Thu Jun 16 12:47:53 2022 -0700

    Chmod task to fix perms

commit f240cbc23f8103fc280a41a15494509529c90b9c
Author: TrainDoctor <11465594+TrainDoctor@users.noreply.github.com>
Date:   Thu Jun 16 12:15:06 2022 -0700

    Working deploy task to send to deck

commit b0d5050896c5ba7a597f82e675c3fc68185eb2cd
Author: TrainDoctor <11465594+TrainDoctor@users.noreply.github.com>
Date:   Tue Jun 14 15:01:10 2022 -0700

    This is broken but new job awaits

commit d67d18ab7766e8dbb79fd48ac347d601f82d3332
Author: TrainDoctor <11465594+TrainDoctor@users.noreply.github.com>
Date:   Tue Jun 14 13:38:29 2022 -0700

    Add README, tasks, moved to pnpm and added an ignore for react (#3)

commit 26dc961f583f66b8333d4f4ddbdba51c00676a7e
Author: AAGaming <aa@bigdumb.gq>
Date:   Wed Jun 1 17:46:56 2022 -0400

    showModal->showContextMenu

commit d8870f20620fc459ed284f652eb204366e6e317d
Author: Jonas Dellinger <jonas@dellinger.dev>
Date:   Mon May 30 22:23:13 2022 +0200

    Update lib, show example of router

commit 82320806496a456a67bd87a844e278c535269b75
Author: AAGaming <aa@bigdumb.gq>
Date:   Sat May 28 23:19:40 2022 -0400

    fix oops

commit 6e73b2796137984e5290a5d5f3e381362f21bfea
Merge: adb8a84 a5d4912
Author: AAGaming <aa@bigdumb.gq>
Date:   Sat May 28 21:52:51 2022 -0400

    Merge branch 'main' of github.com:SteamDeckHomebrew/decky-plugin-template

commit adb8a84d2c313bedd6d6e46408056c3a2677ddb3
Author: AAGaming <aa@bigdumb.gq>
Date:   Sat May 28 21:50:47 2022 -0400

    fix react

commit e264d9754cc0ebbdd8f49994872a85031459d436
Author: AAGaming <aa@bigdumb.gq>
Date:   Sat May 28 21:50:24 2022 -0400

    actually fix react

commit a5d491256fd058796162ef9e234b7b5fea6b6ef0
Author: Jonas Dellinger <jonas@dellinger.dev>
Date:   Thu May 26 13:33:25 2022 +0200

    Use new exported components

commit f554798ee4b92e9e4e7086485fda8ad49ea4bb61
Author: Jonas Dellinger <jonas@dellinger.dev>
Date:   Wed May 25 21:34:31 2022 +0200

    Added support for static assets

commit 76470f49a82ecd8b32344aaf8b0186676d948496
Author: AAGaming <aa@bigdumb.gq>
Date:   Fri May 20 23:50:25 2022 -0400

    readd npm lockfile

commit 569f8b8707636cb52d9b89278c28150d668175db
Author: AAGaming <aa@bigdumb.gq>
Date:   Fri May 20 18:13:31 2022 -0400

    fix dependencies, edit rollup and ts config, switch jsx runtime, use pnpm lockfiles

commit 5a6001026860aca724bf5aa871da93a5edabe12c
Author: Jonas Dellinger <jonas@dellinger.dev>
Date:   Fri May 13 23:18:19 2022 +0200

    Actuall add decky-frontend-lib and remove some old code

commit 52f6ca1c587df2cc74b986bc82419c867c404f47
Author: Jonas Dellinger <jonas@dellinger.dev>
Date:   Sat Apr 23 00:42:11 2022 +0200

    Initial commit

AkazaRenn committed:
Minor, fix Fediverse username

AkazaRenn committed:
Update dependencies

wynn1212 committed:
Update README.md to prepare for project takeover

- Added original author repository information
- Added known issue regarding launching non-Steam games/apps while "Pause on focus loss" is enabled. If at least one game/app is running, non-Steam games/apps will be paused on launch.

wynn1212 committed:
Update README.md formatting

- Placed the solution on a new line for easier reading.

AkazaRenn committed:
Update main.py

Remove unused function

wynn1212 committed:
Merge pull request #4 from AkazaRenn/main

Apply new decky API

wynn1212 committed:
Use step-by-step guide instead of one long sentences

wynn1212 committed:
Enhance step-by-step instructions in Known Issues section for clarity and detail.

wynn1212 committed:
Bump version to 1.0.0, prepare for the big release

- Pause/Resume button is now located on the game icon.
- Migrate settings to json-based.
- Clear `stickyPause` state on manual resume.
- Add `noAutoPauseSet` to select which game shouldn't be paused when on focus loss; this can be toggled on the right side of the Pause Games list.
- Patch `SteamClient.Apps.TerminateApp()` to un-pause games on exit.
- Migrate to Decky V3.

Once again, a big thanks to @AkazaRenn for their contribution.

wynn1212 committed:
Update store screenshot

wynn1212 committed:
Apply subprocess usage fix, though current subprocess usage implementation isn't affected on v1.0.0

wynn1212 committed:
Remove `Pause on Suspend` feature due to suspend hooks removed from steam beta client (#10)

wynn1212 committed:
Change suspend hook suggested by  aarron-lee (#10)

wynn1212 committed:
Bump version to 1.0.1

wynn1212 committed:
update build workflow also include `pause-on-suspend` branch

wynn1212 committed:
update build workflow also include `pause-on-suspend` branch

wynn1212 committed:
Revert "update build workflow also include `pause-on-suspend` branch"

This reverts commit 95bf50b317143c573e7f22779b77d67010d258dd.

wynn1212 committed:
Revert "update build workflow also include `pause-on-suspend` branch"

This reverts commit cf63cefd3c490335e6f7a0c8545965be408aae8e.

aarron-lee committed:
fix: use mobx observable for suspend-resume

aarron-lee committed:
Merge branch 'pause-on-suspend' into mobx_suspend_fix

aarron-lee committed:
fix log for mobx resume

wynn1212 committed:
Merge pull request #11 from aarron-lee/mobx_suspend_fix

fix: use mobx observable for suspend-resume

wynn1212 committed:
Merge branch 'main' into pause-on-suspend

wynn1212 committed:
Revert "Merge branch 'main' into pause-on-suspend"

This reverts commit ed4e931e4794bb973642f83500d8f5f7c3b72510, reversing
changes made to 44430c363eaf1d96f3c0ccdbc74854abab35882a.

wynn1212 committed:
Merge pull request #12 from wynn1212/pause-on-suspend

Merge 'pause-on-suspend' to 'main'

wynn1212 committed:
Fix `@decky/ui` has no exported member`DisplayStatus` and `LifetimeNotification` during build

wynn1212 committed:
Update README.md

Launching another non-steam game/app will remain paused issue is on Steam Client side and has been fixed by Valve.

wynn1212 committed:
Fix steam apps icon not showing due to iconUrl location changed.

wynn1212 committed:
Update icon_hash comment description.

AkazaRenn committed:
Fix QAM offset caused by the latest Steam update

AkazaRenn committed:
Fix marquee not working

wynn1212 committed:
Add Debug message

wynn1212 committed:
Fix resume logic that will actually works on Jul 23 Steam update.

wynn1212 committed:
mbox `m_bResuming` seems to been removed.

wynn1212 committed:
Change suspend resume function logic as toggleable.

wynn1212 committed:
Merge pull request #14 from AkazaRenn/main

Fix QAM offset caused by the latest Steam update

wynn1212 committed:
Merge pull request #16 from wynn1212/dev

Fix Not Resuming on July 22 Steam Client Update

wynn1212 committed:
Update to v1.0.2
